libs/lib_parsing/Parsed_int.ml

```diff
@@ -0,0 +1,119 @@
+(* Brandon Wu
+ *
+ * Copyright (c) 2023 Semgrep Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+(* See the .mli for why this is important.
+ *
+ * In Javascript systems, we choose a 64 implementation for integers, which is
+ * more in line with the native integers used by the Javascript runtime.
+ * This reduces inconsistency in the JS translation.
+ *)
+
+open Ppx_hash_lib.Std.Hash.Builtin
+open Sexplib.Conv
+
+(*****************************************************************************)
+(* Int64 boilerplate *)
+(*****************************************************************************)
+
+(* There isn't an available hash_fold or hash function in Int64.t, so we have to
+   make it up here.
+   Unfortunately, there doesn't seem to be a direct hash function for int64 in
+   the Ppx_hash_lib either, so we just write what it would be.
+*)
+let hash_fold_int64 = Ppx_hash_lib.Std.Hash.fold_int64
+
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+
+(* alt: int64 option? *)
+(* alt: int64 option wrap? *)
+(* could save consumers of the API from dealing with whether the concrete int
+   is representable or not
+*)
+type t = int64 option * Tok.t_always_equal
```

Comment:
hmm, do we need Tok.t_always_equal here? A bit a leaky abstraction then :(

Reply:
i considered leaving it as just `int64 option` (which makes some parts of dealing with the abstract type easier), but overall my impression was that if it is truly a _parsed_ integer, it should be associated to the text that it parsed. The `wrap` almost always appears with the `option`, too.

Reply:
No I meant why not using Tok.t instead of Tok.t_always_equal

Reply:
i mostly didn't want equality of parsed ints to rely on equality of their locations, it felt to me like if you are equating integers, you want the equation of their values